posix.mak

```diff
@@ -147,19 +147,19 @@ endif
 doc: $(DOCS)
 
 $(DOCDIR)/object.html : src/object.d
-	$(DMD) $(DDOCFLAGS) -Df$@ project.ddoc $(DOCFMT) $<
+	$(DMD) $(DDOCFLAGS) -Df$@ project.ddoc $(DOCFMT) $< 2>&1 | grep "Warning:"; test $$? -eq 1
```

Comment:
I think all you need to set the error level is `grep -v "Warning:"`.

Reply:
Also a problem with this pattern is that if dmd fails with no warning the command will succeed.